alerting/src/main/kotlin/org/opensearch/alerting/transport/TransportGetWorkflowAction.kt

```diff
@@ -0,0 +1,134 @@
+import org.opensearch.transport.TransportService
+
+class TransportGetWorkflowAction @Inject constructor(
+    transportService: TransportService,
+    val client: Client,
+    actionFilters: ActionFilters,
+    val xContentRegistry: NamedXContentRegistry,
+    val clusterService: ClusterService,
+    settings: Settings
+) : HandledTransportAction<GetWorkflowRequest, GetWorkflowResponse>(
+    AlertingActions.GET_WORKFLOW_ACTION_NAME, transportService, actionFilters, ::GetWorkflowRequest
+),
+    SecureTransportAction {
+
+    private val log = LogManager.getLogger(javaClass)
+
+    @Volatile override var filterByEnabled = AlertingSettings.FILTER_BY_BACKEND_ROLES.get(settings)
+
+    init {
+        listenFilterBySettingChange(clusterService)
+    }
+
+    override fun doExecute(task: Task, getWorkflowRequest: GetWorkflowRequest, actionListener: ActionListener<GetWorkflowResponse>) {
+        val user = readUserFromThreadContext(client)
+
+        val getRequest = GetRequest(ScheduledJob.SCHEDULED_JOBS_INDEX, getWorkflowRequest.workflowId)
+
+        if (!validateUserBackendRoles(user, actionListener)) {
+            return
+        }
+
+        client.threadPool().threadContext.stashContext().use {
+            client.get(
+                getRequest,
+                object : ActionListener<GetResponse> {
+                    override fun onResponse(response: GetResponse) {
+                        if (!response.isExists) {
+                            log.error("Workflow with ${getWorkflowRequest.workflowId} not found")
+                            actionListener.onFailure(
+                                AlertingException.wrap(
+                                    OpenSearchStatusException(
+                                        "Workflow not found.",
+                                        RestStatus.NOT_FOUND
+                                    )
+                                )
+                            )
+                            return
+                        }
+
+                        var workflow: Workflow? = null
+                        if (!response.isSourceEmpty) {
+                            XContentHelper.createParser(
+                                xContentRegistry, LoggingDeprecationHandler.INSTANCE,
+                                response.sourceAsBytesRef, XContentType.JSON
+                            ).use { xcp ->
+                                workflow = ScheduledJob.parse(xcp, response.id, response.version) as Workflow
+
+                                // security is enabled and filterby is enabled
+                                if (!checkUserPermissionsWithResource(
+                                        user,
+                                        workflow?.user,
+                                        actionListener,
+                                        "workflow",
+                                        getWorkflowRequest.workflowId
+                                    )
```

Comment:
Lets add a `SecureWorkflowMonitorIT` test case for `TransportGetWorkflowAction` also.